src/detection/Psf.cc

```diff
@@ -12,19 +13,16 @@ namespace detection {
 
 namespace {
 
-// Comparison function that determines when we used the cached image instead of recomputing it.
-// We'll probably want a tolerance for colors someday too, but they're just a placeholder right now
-// so it's not worth the effort.
-bool comparePsfEvalPoints(geom::Point2D const &a, geom::Point2D const &b) {
-    // n.b. desired tolerance is actually sqrt(eps), so tolerance squared is eps.
-    return (a - b).computeSquaredNorm() < std::numeric_limits<double>::epsilon();
-}
-
 bool isPointNull(geom::Point2D const &p) { return std::isnan(p.getX()) && std::isnan(p.getY()); }
 
 }  // anonymous
 
-Psf::Psf(bool isFixed) : daf::base::Citizen(typeid(this)), _isFixed(isFixed) {}
+Psf::Psf(bool isFixed, std::size_t capacity)
+  : daf::base::Citizen(typeid(this)),
+  _isFixed(isFixed),
+  _imageCache(new PsfCache(capacity)),
+  _kernelImageCache(new PsfCache(capacity))
```

Comment:
Best practice is to use `std::make_shared` to initialize smart pointers when you're constructing the pointee at the same time (or, if you switch to `std::unique_ptr`, `std::make_unique`).